### PR TITLE
LPS-192379 Added validation that prevents using baseURLs with uppercase letters

### PR DESCRIPTION
**Pull request by jalber786**
**What's new?**
- Added a new condition that checks if the baseURL is incorrect because it contains a uppercase letter.
- Added a new test condition that checks if the aforementioned validation works as expected.
- Added a new test condition that checks the edge-case of 256 characters baseURL.

**What's changed?**
- The baseURL regular expression no longer allows the usage of uppercase characters.
- The tests now use JSONAssert instead of a HTTP call and then an assertion.

See the following **Jira ticket**: [LPS-192379](https://liferay.atlassian.net/browse/LPS-192379?filter=13054)

### CODE REVIEW

**Comment by jalber786** [comment]
ci:test:sf

**Comment by jalber786** [comment]
ci:test:relevant

File: modules/apps/headless/headless-builder/headless-builder-impl/src/main/java/com/liferay/headless/builder/internal/model/listener/APIApplicationRelevantObjectEntryModelListener.java [hunk 1]
```diff
@@ -80,12 +81,25 @@ private void _validate(ObjectEntry objectEntry) {
 					_objectFieldLocalService.getObjectField(
 						objectEntry.getObjectDefinitionId(), "baseURL");
 
+				String message = null;
+				String messageKey = null;
+
+				if (!StringUtil.isLowerCase(baseURL)) {
+					message = "%s must contain only lowercase characters.";
```
Comment (4lejandrito) [inline modules/apps/headless/headless-builder/headless-builder-impl/src/main/java/com/liferay/headless/builder/internal/model/listener/APIApplicationRelevantObjectEntryModelListener.java:88]: The ending `.` is not consisten with the message in the `else` path. Should we remove it?

**Review by 4lejandrito — CHANGES_REQUESTED** [review]
Looks good!

File: modules/apps/headless/headless-builder/headless-builder-impl/src/main/java/com/liferay/headless/builder/internal/model/listener/APIApplicationRelevantObjectEntryModelListener.java (hunk 1 above)
Comment (jalber786) [reply modules/apps/headless/headless-builder/headless-builder-impl/src/main/java/com/liferay/headless/builder/internal/model/listener/APIApplicationRelevantObjectEntryModelListener.java:88]: Hmmm It's strange. If you take a look at the language.properties the message `x-can-have-a-maximum-of-255-alphanumeric-characters` uses a `.` at the end too. So what seems to be inconsistent is the message that was present before. I will change it and send it again!

**Comment by jalber786** [comment]
ci:test:relevant

**Comment by yanan** [comment]
Related failed test are because we're using baseURL with uppercase in the following tests, more details on [compare run results](https://testray.liferay.com/home/-/testray/runs/compare?compare=true&view=details&statusA=3&testrayRunIdB=2407585495&testrayRunIdA=2407011638&statusB=2)
- ListAPIApplicationSchemas.testcase
- EditAPIApplication.testcase
- UnpublishAPIApplication.testcase

**Comment by jalber786** [comment]
ci:test:sf

**Comment by yanan** [comment]
Lots of tests were aborted to run, retrigger now

**Comment by yanan** [comment]
ci:test:relevant

**Comment by 4lejandrito** [comment]
@jalber786 integration tests are failing locally with this PR. Run for example:
```bash
gw testIntegration -a --tests APISortRelevantObjectEntryModelListenerTest
```
Can you have a look?

**Comment by jalber786** [comment]
ci:test:sf

**Comment by jalber786** [comment]
ci:test:relevant

File: modules/apps/headless/headless-builder/headless-builder-test/src/testIntegration/java/com/liferay/headless/builder/model/listener/test/APISchemaRelevantObjectEntryModelListenerTest.java [hunk 2]
```diff
@@ -48,7 +48,9 @@ public void test() throws Exception {
 			JSONUtil.put(
 				"applicationStatus", "unpublished"
 			).put(
-				"baseURL", RandomTestUtil.randomString()
+				"baseURL",
```
Comment (4lejandrito) [inline modules/apps/headless/headless-builder/headless-builder-test/src/testIntegration/java/com/liferay/headless/builder/model/listener/test/APISchemaRelevantObjectEntryModelListenerTest.java:51]: @jalber786 we normally [don't do chaining](https://github.com/liferay/liferay-portal/blob/53c9e2adf7ef70866216d0d3eb48c4bef0fa0939/modules/util/source-formatter/documentation/chaining.markdown) (Brian will likely reject it). Instead do `StringUtil.toLowerCase(RandomTestUtil.randomString()`.

**Review by 4lejandrito — CHANGES_REQUESTED** [review]

**Comment by jalber786** [comment]
ci:test:sf

**Comment by jalber786** [comment]
ci:test:relevant

**Comment by 4lejandrito** [comment]
https://github.com/brianchandotcom/liferay-portal/pull/139412
